R/germline.R

```diff
@@ -92,16 +92,23 @@ germline_single_df <- function(data, reference, species, min_nuc_outside_cdr3, s
     validate_chains_length(min_nuc_outside_cdr3, sample_name) %>%
     rowwise() %>%
     mutate(Germline.sequence = generate_germline_sequence(
-      get("Sequence"), get("V.sequence"), get("J.sequence"),
-      get("V.end"), get("CDR3.start"), get("CDR3.end"), get("J.start"),
-      get("V3.Deletions"), get("J3.Deletions"), sample_name
+      seq = get("Sequence"),
+      v_ref = get("V.sequence"),
+      j_ref = get("J.sequence"),
+      v_end = str_length(get("CDR1.nt")) + str_length("CDR2.nt")
+        + str_length("FR1.nt") + str_length("FR2.nt") + str_length("FR3.nt"),
+      cdr3_start = get("CDR3.start"),
+      cdr3_end = get("CDR3.end"),
+      j_start = get("J.start"),
+      j3_del = get("J3.Deletions"),
+      sample_name = sample_name
     )) %>%
     filter(!is.na(get("Germline.sequence")))
   return(data)
 }
 
-generate_germline_sequence <- function(seq, v_ref, j_ref, v_end, cdr3_start, cdr3_end, j_start, v3_del, j3_del, sample_name) {
-  if (any(is.na(c(seq, v_ref, j_ref, v_end, cdr3_start, cdr3_end, j_start, v3_del, j3_del))) ||
+generate_germline_sequence <- function(seq, v_ref, j_ref, v_end, cdr3_start, cdr3_end, j_start, j3_del, sample_name) {
+  if (any(is.na(c(seq, v_ref, j_ref, v_end, cdr3_start, cdr3_end, j_start, j3_del))) ||
     (seq == "")) {
     warning(
       "Some of mandatory fields in a row ",
```

Comment:
```suggestion
      "Some of mandatory fields in a row are missing",
```

Reply:
This phrase continues on the next lines:
```
    warning(
      "Some of mandatory fields in a row ",
      optional_sample("from sample ", sample_name, " "),
      "contain unexpected NA or empty strings! Found values:\n",
```